src/cli_plugin/install/zip.js

```diff
@@ -0,0 +1,72 @@
+import _ from 'lodash';
+import DecompressZip from '@bigfunger/decompress-zip';
+
+//***********************************************
+//Creates a filter function to be consumed by extractFiles
+//
+//filter: an object with either a files or paths property.
+//filter.files: an array of full file paths to extract. Should match
+//  exactly a value from listFiles
+//filter.paths: an array of root paths from the archive. All files and
+//  folders will be extracted recursively using these paths as roots.
+//***********************************************
+function extractFilter(filter) {
+  if (filter.files) {
+    const filterFiles = filter.files.map((file) => file.replace(/\\/g, '/'));
+    return function filterByFiles(file) {
+      if (file.type === 'SymbolicLink') return false;
+
+      const path = file.path.replace(/\\/g, '/');
+      return !!(_.indexOf(filterFiles, path) !== -1);
+    };
+  }
+
+  if (filter.paths) {
+    return function filterByRootPath(file) {
+      if (file.type === 'SymbolicLink') return false;
+
+      let include = false;
+      filter.paths.forEach((path) => {
```

Comment:
I think [some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) would be more appropriate here, so the iterations stop once a match is found and you don't need to maintain state in a local variable:

```js
return filter.paths.some(path => {
  const regex = new RegExp(`${path}($|/)`, 'i');
  return file.parent.match(regex) && file.type !== 'SymbolicLink';
});
```